docs/developer-docs/security/rust-canister-development-security-best-practices.md

```diff
@@ -461,6 +461,99 @@ Using the rust CDK, the reccuring timer is also lost on upgrade as explained in
 - See the Motoko documentation on [recurringTimer](../../motoko/main/base/Timer.md#function-recurringtimer).
 - See the Rust documentation on [set_timer_interval](https://docs.rs/ic-cdk/0.6.9/ic_cdk/timer/fn.set_timer_interval.html).
 
+## HTTP Outcalls
+
+### Do not store sensitive data (e.g. API keys) in canisters
+
+#### Security concern
+
+Sensitive data is a broad term that varies depending on your application logic and behaviour. Here is a non-exhaustive list of secrets that are typically considered sensitive, such as API keys or tokens:
+* Secrets that allow interaction with non-public endpoints.
+* Secrets that allow querying or modifying endpoints with confidential data.
+* API tokens that are fee-based.
+
+By default, the data stored inside your canister is unencrypted. Therefore if your canister is installed in a malicious replica, it can easily retrieve and steal your keys, tokens, and secrets in plain text.
+
+#### Recommendation
+
+Make sure you don’t store sensitive data inside your canister.
+
+[More information.](https://internetcomputer.org/docs/current/developer-docs/integrations/https-outcalls/https-outcalls-how-it-works#compromised-replicas)
+
+[Data confidentiality security recommendations.](https://internetcomputer.org/docs/current/developer-docs/security/general-security-best-practices#data-confidentiality-on-the-internet-computer)
```

Comment:
For all the links to our own docs: I was told that we should use relative links, here it would be just `general-security-best-practices#data-confidentiality-on-the-internet-computer`

There are some examples in the existing docs and also in #1806